packages/menu-bar/src/vaadin-menu-bar-mixin.js

```diff
@@ -951,31 +956,21 @@ export const MenuBarMixin = (superClass) =>
       const overlay = subMenu._overlayElement;
       overlay.noVerticalOverlap = true;
 
-      this._expandedButton = button;
-
-      requestAnimationFrame(() => {
-        // After changing items, buttons are recreated so the old button is
-        // no longer in the DOM. Reset position target to null to prevent
-        // overlay from closing due to target width / height equal to 0.
-        if (overlay.positionTarget && !overlay.positionTarget.isConnected) {
-          overlay.positionTarget = null;
-        }
```

Comment:
This was needed before we added initial sync rendering in #8290 - back then, the `positionTarget` was only set to `button` after waiting for `updateComplete`. This is no longer needed, we can set it immediately.